yarl/_url.py

```diff
@@ -233,6 +233,8 @@ def build(
             raise ValueError(
                 'Can\'t mix "authority" with "user", "password", "host" or "port".'
             )
+        if port is not None and not isinstance(port, int):
```

Comment:
Can we collapse this?
```suggestion
        if not isinstance(port, (int, NoneType)):
```

Reply:
NameError: name 'NoneType' is not defined

I chose None.__class__ instead.